server/pulp/server/webservices/views/content.py

```diff
@@ -1,13 +1,378 @@
+
+        :return: response containing a serialized list dicts, one for each content type
+        :rtype : django.http.HttpResponse
+        """
+        collection = []
+        cqm = factory.content_query_manager()
+        type_ids = cqm.list_content_types()
+        for type_id in type_ids:
+            link = {'_href': reverse('content_type_resource', kwargs={'type_id': type_id})}
+            link.update({'content_type': type_id})
+            collection.append(link)
+        return generate_json_response(collection)
+
+
+class ContentUnitResourceView(View):
+    """
+    View for individual content units.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, type_id, unit_id):
+        """
+        Return a response containing information about the requested content unit.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param type_id: type of content contained in the repo
+        :type  type_id: string
+        :param unit_id: unique id of a unit
+        :type  unit_id: string
+
+        :return: response containing a dict with data about requested unit
+        :rtype : django.http.HttpResponse
+        """
+        cqm = factory.content_query_manager()
+        try:
+            unit = cqm.get_content_unit_by_id(type_id, unit_id)
+        except MissingResource:
+            msg = _('No content unit resource: %(r)s') % {'r': unit_id}
+            return generate_json_response(msg, response_class=HttpResponseNotFound)
+
+        resource = serialization.content.content_unit_obj(unit)
+        resource.update({'children': serialization.content.content_unit_child_link_objs(resource)})
+        return generate_json_response_with_pulp_encoder(resource)
+
+
+class ContentUnitsCollectionView(View):
+    """
+    View for all content units of a specified type.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, type_id):
+        """
+        Return a response with a serialized list of the content units of the specified type.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param type_id: the list of content units will be limited to this type
+        :type  type_id: string
+
+        :return: response with a serialized list of dicts, one for each unit of the type.
+        :rtype: django.http.HttpResponse
+        """
+        cqm = factory.content_query_manager()
+        all_units = cqm.find_by_criteria(type_id, Criteria())
+        all_processed_units = []
+        for unit in all_units:
+            unit = serialization.content.content_unit_obj(unit)
+            unit.update({'_href': '/'.join([request.get_full_path().rstrip('/'),
+                                            unit['_id'], ''])})
+            unit.update({'children': serialization.content.content_unit_child_link_objs(unit)})
+            all_processed_units.append(unit)
+
```

Comment:
I removed many of the extra lines, but left some (including this one) since it follows a block and it is easier for me to read with the space.